riscv/src/compiler.rs

```diff
@@ -1416,44 +1324,27 @@ fn process_instruction(instr: &str, args: &[Argument], coprocessors: &CoProcesso
             vec![format!("x1 <== jump_dyn({rs});")]
         }
         "call" | "tail" => {
-            // Depending on what symbol is called, the call is replaced by a
-            // powdr-asm call, or a call to a coprocessor if a special function
-            // has been recognized.
             assert_eq!(args.len(), 1);
             let label = &args[0];
-            let replacement = match label {
-                Argument::Expression(Expression::Symbol(l)) => {
-                    try_coprocessor_substitution(l, coprocessors)
-                }
-                _ => None,
-            };
-            match (replacement, instr) {
-                (None, instr) => {
-                    let arg = argument_to_escaped_symbol(label);
-                    let dest = if instr == "tail" { "tmp1" } else { "x1" };
-                    vec![format!("{dest} <== jump({arg});")]
-                }
-                // Both "call" and "tail" are pseudo-instructions that are
-                // supposed to use x6 to calculate the high bits of the
-                // destination address. Our implementation does not touch x6,
-                // but no sane program would rely on this behavior, so we are
-                // probably fine.
-                (Some(replacement), "call") => vec![replacement],
-                (Some(replacement), "tail") => {
-                    vec![replacement, "tmp1 <== jump_dyn(x1);".to_string()]
-                }
-                (Some(_), _) => unreachable!(),
-            }
+            let arg = argument_to_escaped_symbol(label);
+            let dest = if instr == "tail" { "tmp1" } else { "x1" };
+            vec![format!("{dest} <== jump({arg});")]
         }
         "ecall" => {
             assert!(args.is_empty());
-            vec!["x10 <=X= ${ std::prover::Query::Input(std::convert::int(std::prover::eval(x10))) };".to_string()]
+            // save ra/x1
+            push_register("x1")
+                .into_iter()
+                // jump to to handler
+                .chain(std::iter::once("x1 <== jump(__ecall_handler);".to_string()))
+                // restore ra/x1
+                .chain(pop_register("x1"))
+                .collect()
         }
         "ebreak" => {
             assert!(args.is_empty());
-            // This is using x0 on purpose, because we do not want to introduce
-            // nondeterminism with this.
-            vec!["x0 <=X= ${ std::prover::Query::PrintChar(std::convert::int(std::prover::eval(x10))) };\n".to_string()]
+            // we don't use ebreak for anything, ignore
```

Comment:
Shouldn't we panic? Is his used anywhere?

Reply:
yes, its used in the risc-v asm instruction IRC, for example

Reply:
Maybe at least prevent witgen from succeeding if this is executed?

Reply:
I would prefer removing/modifying the instruction test, because I think this can be dangerous.

Reply:
How would a `noop` in place of an `ebreak` be dangerous? This was previously used to call prover print_char.
The tests that use `ebreak` are those generated from the risc-v test suite...
Should I open an issue?